blocks/top-banner/top-banner.js

```diff
@@ -39,12 +39,26 @@ export default function decorate(block) {
   if (variant === 'dismissible') {
     closeButton = button({ class: 'top-banner-close', 'aria-label': 'Close banner' });
     closeButton.addEventListener('click', () => {
+      const bannerHeight = block.offsetHeight;
+
+      block.style.maxHeight = `${bannerHeight}px`;
+      // Force a reflow
+      block.offsetHeight; // eslint-disable-line no-unused-expressions
+
       block.classList.add('dismissed');
       document.body.classList.remove('has-top-banner');
-      // Use setTimeout to remove the banner after the transition
+
+      document.body.style.paddingTop = '0';
+      const header = document.querySelector('header');
+      if (header) {
+        header.style.top = '0';
+      }
+
+      // Remove the inline max-height after the transition
       setTimeout(() => {
-        block.remove();
-      }, 300); // 300ms matches the transition duration in CSS
+        block.style.maxHeight = '';
+      }, 500); // 500ms matches the transition duration in CSS
```

Comment:
I thought maybe the code was going to look something like:
```
addEventListner('click', () => {
   documnet.querySelector('. top-banner').remove();
   sessionStorage.setItem('topBannerDismissed', 'true');
})
```

I'm surprised to see that we're playing with the body style's at all.  Just nuke the banner and allow the dom to reposition itself?

Reply:
see my comment below